Rickify

mitechie opened the pull request:
- Update the longrunning scripts to be a cli script
- Use a sqlite db to store the parsed data from the log
- Track the logs that have been ingested for data so we can follow up with loading only new files
- Add a makefile that provides guided steps to getting things setup and loaded
- Add a readme that walks the make steps in the right order

to QA:
see if you can use the readme and makefile to successfully load the log files and generate the longrunning stats

mitechie committed:
Add support for cloud/region/juju version data.

- Add a get.sh which helps me loop/pull down log files
- Update to deal with zipped log files automatically
- Refactor the longrunning to check full lines via regex vs each part of a
string
- Add support for cloud the uuid is on
- Add support for region the uuid is on
- Add support for Juju controller version the uuid is on

Update the sourcing of novarc

mitechie committed:
Correct clouds counting and pre-2 output

mitechie committed:
Add a workloads searching file, string hard coded atm

mitechie committed:
Minor refactor for longrunning

mitechie committed:
Add total UUIDs seen, bug in get.sh

mitechie committed:
Update to parse the region/etc info on a week by week basis.
Update to use the date as a week counter so that we can roll over into the new
year better.

mitechie committed:
Add a new SQL based long running script

mitechie committed:
Add the Makefile I've started

mitechie committed:
Fit linting

mitechie committed:
Fix the bug in the logs path

mitechie committed:
Add output of the version counts

mitechie committed:
Add version and clouds to the output

mitechie committed:
Add cloud output to the normal data run.

- Show the list of clouds with counts for the latest day.
- Show the break down of regions in those clouds for the latest day.

mitechie committed:
Update to move to more click cli components

mitechie committed:
Fix the rest of the makefile running all longrunning data

mitechie committed:
Update makefile to get-logs working

mitechie committed:
Update the readme on how to get started

mitechie committed:
Update the readme on how to get started

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+venv: bin/python
+bin/python:
+	# needs python3-dev to build keystoneclient deps
+	virtualenv -p /usr/bin/python3 .
```
You morally opposed to hiding the venv in .venv?  /me hates seeing those venv artifacts everywhere.

mitechie replied:
No, guess not. just that it's harder to source when you can't see the directory. the bin/etc is hidden. Updated.

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+
+.PHONY: get-logs
+get-logs: logs/api/1 logs/api/2 check-swift
+	swift list production-juju-ps45-cdo-jujucharms-machine-1.canonical.com | grep  201 | grep api.jujucharms.com.log > logs/api/logs1.list
```
this is using system swift not the one from the venv

bac commented on logs/api/get.sh:
```diff
@@ -0,0 +1,23 @@
+# swift list production-juju-ps45-cdo-jujucharms-machine-1.canonical.com G 201610 G api.jujucharms.com.log > logs1.list
+# swift list production-juju-ps45-cdo-jujucharms-machine-2.canonical.com G 201610 G api.jujucharms.com.log > logs2.list
+
+source $HOME/archive-anon-cdo-jujucharms-ro.novarc
```
Rather than sourcing the novarc file again, especially since you aren't really sure where everyone put there's, I'd just expect that it is already sourced and do a check to ensure at least one of the variables is set, like:

```
ERROR="Need to provide NOVA_PASSWORD"
: ${NOVA_PASSWORD:?$ERROR}
```

mitechie replied:
Ah, I added that to the makefile but not to this script. Thanks. I'll update it.

bac commented on long-running/longrunning.py:
```diff
@@ -1,34 +1,371 @@
+
+
+def connect_sql():
+    conn = sqlite3.connect('models.db')
```
use DB_NAME

mitechie replied:
ty

bac commented on long-running/longrunning.py:
```diff
@@ -1,34 +1,371 @@
+
+
+def recreate_db():
+    """Simple program that greets NAME for a total of COUNT times."""
```
Typo? Greets?

bac replied:
What are NAME and COUNT?

bac commented on README:
```diff
@@ -0,0 +1,16 @@
+  make
+  source ~/path/to/novarc/for/swift
+  make get-logs
+  make _initdb
```
Why the underscore? It prevents `make` from doing tab completion.

mitechie replied:
Yes, the focus is that this should only happen once and be something very manually done as a setup step. I'm adding a "update-db" one to be run that is safe to run when logs are updated/etc.

bac commented on long-running/README:
```diff
@@ -0,0 +1,11 @@
+In order to load the long running model data you need access to the apache
+logs for Jujucharm.com and there are two servers there.
+
```
I find this README a bit confusing as it is in the `longrunning` subdirectory but the commands specified need to be run in the parent as there is no Makefile here.

It may be less confusing to just include anything you need to say about `longrunning` in the parent README.

bac commented on README:
```diff
@@ -0,0 +1,16 @@
+  source ~/path/to/novarc/for/swift
+  make get-logs
+  make _initdb
+  make longrunning
```
OK, I got this far. Everything worked.

BUT...

what the hell was the point? What am I supposed to see? Can you beef up the README to describe in better detail what the project is for, how to see the results, etc?

bac replied:
This comment was just plain dumb. Sorry I somehow missed all of the fine information that scrolled past when running `make longrunning`.

mitechie replied:
Updated the readme. Thanks for the outsider pov

bac reviewed (changes requested):
Looks good Rick but see my suggestions.

bac commented:
Also please add a .gitignore. After running I see:

```
[qa-1]bac@salchicha:~/git/gypsy-danger$ git status
On branch qa-1
Your branch is ahead of 'master' by 18 commits.
  (use "git push" to publish your local commits)
Untracked files:
  (use "git add <file>..." to include in what will be committed)

	bin/
	include/
	lib/
	models.db
	pip-selfcheck.json
	share/
```

bac commented:
QA-OK

mitechie committed:
Update the readme with more info per review request

mitechie committed:
Updates per review

- update .gitignore
- move venv to a hidden directory
- update makefile for ^
- Improve the readme file
- clean up some garbage in the longrunning.py file

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+clean-all: clean_venv
+	if [ -d dist ]; then \
+		rm -r dist; \
+    fi
```
Is this a tab issue? May want to try it.

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+	.venv/bin/pip install flake8
+
+lint: .venv/bin/flake8
+	flake8 long-running
```
This is not using the venv flake8

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+.venv/bin/python:
+	# needs python3-dev to build keystoneclient deps
+	virtualenv -p /usr/bin/python3 .venv
+	.venv/bin/pip install click
```
$(PIP)

bac commented on Makefile:
```diff
@@ -0,0 +1,79 @@
+
+
+.venv/bin/flake8: .venv
+	.venv/bin/pip install flake8
```
$(PIP)

bac reviewed (commented):
Sorry I found a couple more things. Perhaps for your next branch.
